beacon_node/network/src/beacon_processor/mod.rs

```diff
@@ -1050,12 +1062,28 @@ impl<T: BeaconChainTypes> BeaconProcessor<T> {
                 reprocess_work_rx: ready_work_rx,
             };
 
+            let disable_backfill_rate_limiting = chain.config.disable_backfill_rate_limiting;
+
             loop {
                 let work_event = match inbound_events.next().await {
                     Some(InboundEvent::WorkerIdle) => {
                         self.current_workers = self.current_workers.saturating_sub(1);
                         None
                     }
+                    Some(InboundEvent::WorkEvent(event))
+                        if !disable_backfill_rate_limiting && event.is_backfill() =>
+                    {
+                        if let Err(e) = work_reprocessing_tx
+                            .try_send(ReprocessQueueMessage::BackfillSync(event))
+                        {
+                            error!(
+                                self.log,
+                                "Unable to queue backfill work event";
+                                "error" => %e
+                            )
+                        }
```

Comment:
thanks for the explanation, I'll look into this

Reply:
Just to clarify, is this what you meant by sending a message to sync?

```rust
let Work::ChainSegment { ref process_id, .. } = event.work;
let message = SyncMessage::BatchProcessed {
    sync_type: process_id.clone(),
    result: BatchProcessResult::NonFaultyFailure,
};
self.sync_tx.send(message).unwrap_or_else(|e| {
    debug!(self.log, "Could not send message to the sync service";
"error" => %e)
});
```

Just navigating through the code and looks like sending a `NonFaultyFailure` would trigger a retry on downloading the batch.

I'm wondering whether skipping the reprocess queue (if its full) and just let `BeaconProcessor` process / queue the batch be an alternative option? since this won't require re-downloading the batch from a peer, it just means we're falling back to the existing behaviour with no rate-limiting. Although this feels like a edge case to me, as the delay queue default capacity is really high (`1073741823`), so as long as it's handled one of these ways I don't really have a strong preference

Reply:
going to the previous path sounds like a good option. And yeah the queue is very big, but I think we've seen it fill at least once. Of course here we are, like you said, handling highly unlikely cases

Reply:
Done! Falling back to existing path if  it fails to queue the batch.